### PR TITLE
[OpenAI] Regenerated with sync stack option enabled

### PR DESCRIPTION
**Pull request by mssfang**
Added OpenAI Code owners.

Regenerated OpenAI with sync stack feature.

Steps:

Add `enable-sync-stack: true` to the `tspconfig.yaml` under the line `partial-update: true`.

### CODE REVIEW

File: sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/generated/OpenAIClientTestBase.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.test.TestBase;
+import com.azure.core.test.TestMode;
+import com.azure.core.util.Configuration;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import java.time.OffsetDateTime;
+import reactor.core.publisher.Mono;
+
+class OpenAIClientTestBase extends TestBase {
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/generated/OpenAIClientTestBase.java:20]: @srnagar Do you know if there is an option to exclude `generated` test folder.

File: sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/OpenAIClientTestBase.java [hunk 2]
```diff
@@ -97,7 +97,6 @@ static void assertChoices(int[] index, CompletionsLogProbabilityModel[] logprobs
     static void assertChoice(int index, CompletionsLogProbabilityModel logprobs,
                              CompletionsFinishReason finishReason, Choice actual) {
         assertNotNull(actual.getText());
-        assertNotNull(actual.getFinishReason());
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/OpenAIClientTestBase.java:100]: turn out the finishReason not guarantee to be NonNull.

**Comment by brandom-msft** [comment]
Got my permissions sorted, does Codeowners file still complain?

File: sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/generated/OpenAIClientTestBase.java (hunk 1 above)
Comment (srnagar) [reply sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/generated/OpenAIClientTestBase.java:20]: Adding `generate-tests: false` to tspconfig.yaml will exclude the generated test folder.

Comment (mssfang) [reply sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/generated/OpenAIClientTestBase.java]: it works.

**Comment by mssfang** [comment]
> Got my permissions sorted, does Codeowners file still complain?

It is still complain but will need to request permission to join cognitive write group

**Review by jpalvarezl — APPROVED** [review]

**Pull request merged**